src/Modal/Modal.tsx

```diff
@@ -1,11 +1,23 @@
 /** @jsx jsx */
-import { jsx, css } from "@emotion/core";
+import { jsx, css, ClassNames } from "@emotion/core";
 import React, { useEffect } from "react";
 import * as typography from "../typography";
 import { colors } from "../colors";
 import * as CSS from "csstype";
+import classnames from "classnames";
 
 interface Props {
+  /**
+   * Override the the default element used to render the outermost component
+   *
+   * All props provided will be merged with props that this component adds,
+   * including `className`s being merged. You can use this to make the modal a
+   * `<form>`
+   *
+   * @default `<div />`
+   */
+  as?: React.ReactElement;
```

Comment:
Did we want to include ` | keyof JSX.IntrinsicElements` here to be consistent?

Reply:
Good point. We're using `as?: React.ReactElement;` everywhere we define `as`, and we're using `inputAs?: React.ReactElement | keyof JSX.IntrinsicElements;` in `TextField`.

Let's clean this up in a feature release soon.